spec/models/manageiq/providers/hawkular/middleware_manager/alert_profile_manager_spec.rb

```diff
@@ -1,6 +1,12 @@
 describe ManageIQ::Providers::Hawkular::MiddlewareManager::AlertProfileManager do
   let(:client) { double('Hawkular::Alerts') }
-  let(:subject) { described_class.new(client) }
+  let(:stubbed_ems) do
+    ems = instance_double('::ManageIQ::Providers::Hawkular::MiddlewareManager',
```

Comment:
This smells like `ask, don't tell` to me. Maybe you should either receive the id_prefix as an argument, or move the code related to the usage of this to MiddlewareManager itself.